src/Persistence/Yaml/SerializerBuilderExtensions.cs

```diff
@@ -32,4 +33,21 @@ public static DeserializerBuilder WithFirstClassModels(this DeserializerBuilder
             })
             .WithTypeConverter(new ControlPropertiesCollectionConverter());
     }
+
+    private static TBuilder AddAttributeOverrides<TBuilder>(this TBuilder builder)
+       where TBuilder : BuilderSkeleton<TBuilder>
+    {
+        var types = typeof(Control).Assembly.DefinedTypes;
+        foreach (var type in types)
```

Comment:
Why do we still need this?

Reply:
this helps keeping the `Control` class persistence-agnostic, so we don't pollute it with attributes specific to the file format (eg. YAML)

Reply:
I don't think adding attributes is polluting. I am thinking injecting attributes at runtime is very unexpected behavior

Reply:
it's more declarative. EntityFramework does something similar when building the config for the entities definition. 
The real problem that we want to solve is avoiding having configuration/attributes specific to a file format (YAML in our case) embedded into our POCO classes. It would lead to the same problem we have right now on DocServer with JSON

Reply:
Control class is designed for persistence. It is only intended to hold data during read/write operations.

Reply:
that's correct, but we should avoid having details specific to the file format in these classes